hashintel/hash
# H-2194, H-1777: add `/admin/users` page where hash instance admins can set the feature flags of a user

## 🌟 What is the purpose of this PR?




This PR adds `/admin/users` and `/admin/users/:uuid` pages, so that the enabled feature flags of users can be modified by HASH instance admins. The `/admin/user` page can be navigated to via the sidebar and the account dropdown (H-1777)

## 🔗 Related links





- H-2194
- H-1777

## 🔍 What does this change?




- adds a migration which gives the instance admin account group edit permissions on all existing user entities
- when creating a new user, gives the instance admin account group edit permissions on the new user entity

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



Manual testing

## ❓ How to test this?



1. Login as an instance admin
2. Navigate to `/admin/users` via the sidebar or the account dropdown
3. Modify the enabled feature flags for an existing user
4. Ensure they persisted correctly (by refreshing the page for example)

## 📹 Demo



https://github.com/hashintel/hash/assets/42802102/def23b9a-46b1-49c9-9b12-59f3eb496421
